scripts/lib_refactor/README.md

```diff
@@ -0,0 +1,36 @@
+# hal_gigadevice lib refactor
+
+This folder contains `hal_gigadevice` lib refactor scripts.
+
+## Dependencies
+
+```
+apt install fish
+```
+
+## Usage
+
+1. Download `GD32xxxx Firmware Library` from [gigadevice](http://www.gd32mcu.com/en/download/7)
+2. make folders name like `gd32c10x` at root of `hal_gigadevice` and decompress file to this folder.
+
+files in folder should like this:
+``` shell
+gd32c10x/
+├── Examples
+├── Firmware
+├── GD32C10x_Firmware_Library_V1.0.2
+├── Template
+└── Utilities
+```
+
+3. use folder as parameter to execute `scripts/lib_refactor/make_new_hal.sh`
+
+## Example
+
+``` shell
+wget -O /tmp/gd32c10x.rar http://www.gd32mcu.com/download/down/document_id/216/path_type/1
+mkdir gd32c10x
+unrar x /tmp/gd32c10x.rar gd32c10x
+mv gd32c10x/*/* gd32c10x/
+scripts/lib_refactor/make_new_hal.sh gd32c10x
```

Comment:
`lib_refactor` isn't probably a good name if it adds scripts to add a new HAL. I think that we should just provide a script to add a HAL (or update, which means remove and add), such script should also run patch scripts under the hood. 

In fact, all the steps above could be handled by the script itself (download, uncompress, copy/delete, patch).

Reply:
I made a preliminary version and wanted to check if it meets your expectations.
Contains download, decompression, patch functions.

I'm not sure if it makes sense to remove hal.